src/interpreter/ByteCodeInterpreter.cpp

```diff
@@ -1760,8 +1760,23 @@ NEVER_INLINE void InterpreterSlowPath::resolveNameAddress(ExecutionState& state,
     while (env) {
         auto result = env->record()->hasBinding(state, code->m_name);
         if (result.m_index != SIZE_MAX) {
-            registerFile[code->m_registerIndex] = Value(count);
-            return;
+            // NOTE checking unscopables on hasBinding is impossible.
+            // because storeByNameWithAddress needs to call hasBinding also.
+            // If envRec.[[IsWithEnvironment]] is false, return true. skipped
+            // ObjectEnvironmentRecord is created only for with statement, so `IsWithEnvironment` is always true.
+            bool foundUnscopables = false;
+            if (env->record()->isObjectEnvironmentRecord()) {
+                ObjectPropertyName propertyName(code->m_name);
+                auto obj = env->record()->asObjectEnvironmentRecord()->bindingObject();
+                Value unscopables = env->record()->asObjectEnvironmentRecord()->bindingObject()->get(state, ObjectPropertyName(state.context()->vmInstance()->globalSymbols().unscopables)).value(state, obj);
```

Comment:
It seems like that `env->record()->asObjectEnvironmentRecord()->bindingObject()->get` can be simply replaced with `obj->get`

Reply:
I see